subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaInstallationRegistry.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.provider.Provider;
+
+import java.io.File;
+
+/**
+ * Allows information about Java installations to be queried.
+ *
+ * <p>An instance of this service is available for injection into tasks, plugins and other types.
```

Comment:
(I haven't looked all the way through this PR) -- would this be available to worker actions too?

Reply:
It would be, but as a later step.